src/createPackage/models/createPackageManager.ts

```diff
@@ -165,6 +191,132 @@ export class CreatePackageManager {
     return jobCreated;
   }
 
+  public async createPackageRoi(userInput: ICreatePackageRoi): Promise<ICreateJobResponse | ICallbackExportResponse> {
+    const { dbId, crs, priority, callbackURLs } = userInput;
+    let roi = userInput.roi;
+    const layer = await this.rasterCatalogManager.findLayer(userInput.dbId);
+    const layerMetadata = layer.metadata;
+    if (!roi) {
+      // convert and wrap layer's footprint to featureCollection
+      const layerFeature = feature(layerMetadata.footprint as Geometry);
+      roi = featureCollection([layerFeature]);
+    }
+
+    let { productId: resourceId, productVersion: version, productType } = layerMetadata;
+    const featuresRecords = this.parseFeatureCollection(roi, layerMetadata);
+    const tileEstimatedSize = this.getTileEstimatedSize(layerMetadata.tileOutputFormat as TileOutputFormat);
+
+    resourceId = resourceId as string;
+    version = version as string;
+    productType = productType as ProductType;
+
+    const srcRes = layerMetadata.maxResolutionDeg as number;
+    const maxZoom = degreesPerPixelToZoomLevel(srcRes);
+
+    // ROI vs layer validation section - zoom + geo intersection
+    featuresRecords.forEach((record) => {
+      if (record.zoomLevel > maxZoom) {
+        throw new BadRequestError(`The requested resolution ${record.targetResolution} is larger then then product resolution ${srcRes}`);
+      }
+
+      // generate sanitized bbox for each original feature
+      record.sanitizedBox = this.sanitizeBbox(
+        record.geometry as Polygon | MultiPolygon,
+        layerMetadata.footprint as Polygon | MultiPolygon,
+        record.zoomLevel
+      );
+      if (!record.sanitizedBox) {
+        throw new BadRequestError(
+          `Requested ${JSON.stringify(record.geometry as Polygon | MultiPolygon)} has no intersection with requested layer ${
+            layer.metadata.id as string
+          }`
+        );
+      }
+    });
+    const layerBbox = PolygonBbox(roi); // bounding box of entire ROI
+
+    const dupParams: JobExportDuplicationParams = {
+      resourceId,
+      version,
+      dbId,
+      roi,
+      crs: crs ?? DEFAULT_CRS,
+    };
+
+    const callbacks = callbackURLs.map((url) => <ICallbackTargetExport>{ url, roi });
+    const duplicationExist = await this.checkForExportDuplicate(dupParams, callbacks);
```

Comment:
shouldn't we check for duplications at the beginning of this method?

Reply:
its the earlier position to check the duplication, becouse on first stage it will normolize  the request data, interpulate the geo and validating the actual request are legal (not bad request type) and then only after understanding the request ok, than it will check if already there is duplication

Reply:
done, after converting type